File: src/dataset/Java/anagram_exercism-ThomasZumsteg/claude/ClaudeSonnet4_Anagram_v3.java
import java.util.ArrayList;
import java.util.List;
import java.util.Arrays;

public class Anagram {
    private final String anagram;
    private final char[] sortedAnagramLetters;
    private final String anagramLower;

    public Anagram(String anagram) {
        this.anagram = anagram;
        this.anagramLower = anagram.toLowerCase();
        this.sortedAnagramLetters = this.anagramLower.toCharArray();
        Arrays.sort(this.sortedAnagramLetters);
    }

    public List<String> match(List<String> words) {
        List<String> anagrams = new ArrayList<>(words.size());

        for (String word : words) {
            if (word.length() != anagram.length()) {
                continue;
            }

            String wordLower = word.toLowerCase();
            
            if (anagramLower.equals(wordLower)) {
                continue;
            }

            char[] wordLetters = wordLower.toCharArray();
            Arrays.sort(wordLetters);

            if (Arrays.equals(sortedAnagramLetters, wordLetters)) {
                anagrams.add(word);
            }
        }
        return anagrams;
    }
}

File: src/dataset/python/anagram_exercism-ThomasZumsteg/claude/ClaudeSonnet4_anagram_v3.py
"""Finds anagrams of a word in a list of words"""

def detect_anagrams(anagram, word_list):
    """Finds any words in [word_list] that are anagrams of [anagram]"""
    anagram_lower = anagram.lower()
    anagram_sorted = sorted(anagram_lower)
    anagram_len = len(anagram_lower)
    
    result = []
    for word in word_list:
        word_lower = word.lower()
        if (len(word_lower) == anagram_len and 
            word_lower != anagram_lower and 
            sorted(word_lower) == anagram_sorted):
            result.append(word)
    
    return result